File: src/main/java/ru/highloadcup/check/LocationEOCustomChecker.java
package ru.highloadcup.check;

import org.springframework.stereotype.Component;
import ru.highloadcup.api.LocationEO;

@Component
public class LocationEOCustomChecker implements CustomChecker<LocationEO> {
    @Override
    public void checkFullyNull(LocationEO object) {
        if (object.getCity() == null
                && object.getCountry() == null
                && object.getDistance() == null
                && object.getPlace() == null) {
            throw new IllegalArgumentException("object is fully null");
        }
    }
}

File: src/main/java/ru/highloadcup/check/LocationEOCustomChecker.kt
package ru.highloadcup.check

import org.springframework.stereotype.Component
import ru.highloadcup.AllOpen
import ru.highloadcup.api.LocationEO

@AllOpen
@Component
class LocationEOCustomChecker : CustomChecker<LocationEO> {
    override fun checkFullyNull(checkingObject: LocationEO) {
        if (checkingObject.city == null
                && checkingObject.country == null
                && checkingObject.distance == null
                && checkingObject.place == null) {
            throw IllegalArgumentException("checkingObject is fully null")
        }
    }
}
